plugin/build.gradle.kts

```diff
@@ -31,3 +33,23 @@ gradlePlugin {
         }
     }
 }
+
+afterEvaluate {
+    fun PomFilterContainer.removeTestDependencies() {
+        pom.whenConfigured {
+            dependencies.removeIf {
+                // Stolen from JetBrains' own sample at
+                // https://github.com/JetBrains/kotlin/blob/v1.2.50/buildSrc/src/main/kotlin/plugins/PublishedKotlinModule.kt#L86
+                InvokerHelper.getMetaClass(it).getProperty(it, "scope") == "test"
+            }
+        }
+    }
+
+    (tasks["uploadArchives"] as? Upload?)?.let {
+        (it.repositories["mavenDeployer"] as PomFilterContainer).removeTestDependencies()
+    }
+
+    (tasks["install"] as? Upload?)?.let {
```

Comment:
Same here

Reply:
Nice, so we can probably combine it into one statement. I'll try